tests/schedulers/test_direct.py

```diff
@@ -70,3 +70,41 @@ def test_submit_script_with_num_cores_per_mpiproc(scheduler, template):
     )
     result = scheduler.get_submit_script(template)
     assert f'export OMP_NUM_THREADS={num_cores_per_mpiproc}' in result
+
+
+def run_sleep_100():
+    """Util function for `test_kill_job`. Has to be outside of test to be pickable."""
+    import subprocess
+
+    subprocess.Popen(['sleep', '100'])
+
+
+@pytest.mark.timeout(timeout=10)
+def test_kill_job(scheduler):
```

Comment:
The test I added, increases on my machine the testing time roughly by 1.7 s (test direct was before 2.3 seconds). Spawning and killing needs some time therefore the while loops. I already have set the start method to `spawn` so less resources are used. Not sure how to otherwise decrease the time of the test.

Reply:
Realized that forking is actually much faster, now test is in normal performance regime. Also makes handling much easier.